gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -108,4 +123,97 @@ def run(
         for i in range(num_images):
             if wRi_list[i] is not None and wti_values.exists(i):
                 wti_list[i] = wti_values.atPoint3(i)
-        return wti_list
+
+        # Compute the metrics.
+        if gt_wTi_list is not None:
+            ta_metrics = _compute_metrics(inliers, outliers, i2Ui1_dict, wRi_list, wti_list, gt_wTi_list)
+        else:
+            ta_metrics = None
+
+        return wti_list, ta_metrics
+
+
+def _get_measurement_angle_errors(
+    i1_i2_pairs: Tuple[int, int],
+    i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
+    gt_i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
+) -> List[float]:
+    """Returns a list of the angle between i2Ui1_measurements and gt_i2Ui1_measurements for every 
+    (i1, i2) in i1_i2_pairs.
+
+    Args:
+        i1_i2_pairs: List of (i1, i2) tuples for which the angles must be computed.
+        i2Ui1_measurements: Measured translation direction of i1 WRT i2.
+        gt_i2Ui1_measurements: Ground truth translation direction of i1 WRT i2.
+
+    Returns:
+        List of angles between the measured and ground truth translation directions.
+    """
+    errors = []
+    for (i1, i2) in i1_i2_pairs:
+        if (i1, i2) in i2Ui1_measurements and (i1, i2) in gt_i2Ui1_measurements:
+            errors.append(
+                comp_utils.compute_relative_unit_translation_angle(
+                    i2Ui1_measurements[(i1, i2)], gt_i2Ui1_measurements[(i1, i2)]
+                )
+            )
+    return errors
+
+
+def _compute_metrics(
+    inlier_i1_i2_pairs: List[Tuple[int, int]],
+    outlier_i1_i2_pairs: List[Tuple[int, int]],
+    i2Ui1_dict: Dict[Tuple[int, int], Optional[Unit3]],
+    wRi_list: List[Optional[Rot3]],
+    wti_list: List[Optional[Point3]],
+    gt_wTi_list: List[Optional[Pose3]],
+) -> GtsfmMetricsGroup:
+    """Computes the translation averaging metrics as a metrics group.
+
+    Args:
+        w_i2Ui1_inlier_measurements: Inlier 1DSfM translation direction measurements in world frame.
+        w_i2Ui1_outlier_measurements: Outlier 1DSfM translation direction measurements in world frame.
+        wRi_list: Estimated camera rotations from rotation averaging.
+        wti_list: Estimated camera translations from translation averaging.
+        gt_wTi_list: List of ground truth camera poses.
+
+    Returns:
+        Translation averaging metrics as a metrics group. Includes the following metrics:
+        - Number of inlier, outlier and total measurements.
+        - Distribution of translation direction angles for inlier measurements.
+        - Distribution of translation direction angle for outlier measurements.
```

Comment:
should we also add the precision/recall for 1dsfm, if given GT poses?

Reply:
To do that we need a threshold on the input measurement angle error to define what constitutes TP vs FP. We could do that, but this signal is kind of observed from the inlier and outlier errors.

Reply:
This seems to be the only unresolved comment, I will add them too. Have the changes locally but I'd let CI run overnight to check the deep front end metrics.

Reply:
marking as resolved, thanks